extensions/amp-list/0.1/amp-list.js

```diff
@@ -1200,6 +1200,7 @@ export class AmpList extends AMP.BaseElement {
       urlReplacement: this.getPolicy_(),
       refresh,
       token,
+      stripPrefix: this.element.getAttribute('strip-prefix') || undefined,
```

Comment:
What do you think about reading `strip-prefix` in `batchFetchJsonFor` as opposed to passing it in from the `amp-list` call ? It could be implemented similarly to `credentials` here:

https://github.com/ampproject/amphtml/blob/f03b2789cb18249569852f8124ac25c8c0455066/src/batched-json.js#L119-L121

This would make it more straight forward for expanding support--excluding the attribute to `amp-list` or expanding to other components would only be a change in the validation rules.

Reply:
I had done it that way originally. @jridgewell requested I do it this way for the explicitness. I don't particularly care one way or the other.

> excluding the attribute to amp-list or expanding to other components would only be a change in the validation rules.

That would be true for any component that uses `batched-json`, but many amp components go directly through `xhr-impl.js` (e.g. `amp-consent`).  Each of those components would still need component-specific code.

Reply:
I don't think it necessarily needs to be in `batched-json` but in general having `strip-prefix` as a common attribute makes more sense to me. It seems to me that there is reasonable likelihood of expanding beyond `amp-list` and `amp-consent`, i.e. this could be relevant to `amp-autocomplete`, `amp-form`, `amp-state`, etc.

Reply:
Sure. I could change the argument signature to:

```jsdoc
/**
 * @param {!Response} res fetch response to convert to json.
 * @param {Element|undefined} element whose fetch we may want to strip prefix for
 */
function xssiJson(res, el ) { ... }
```

---

The benefit of passing the element in is that we can more easily force each amp component to have the same attr name (strip-prefix).  The con is that the testing surface becomes much larger (a whole `Element` for this fn instead of just a String).

Let me know if you feel strongly about this and I'll change it, but otherwise I'd like to leave it as-is for this PR because I know there are others that think we shouldn't pass the element in.